src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/PackageManagerUICommandHandler.cs

```diff
@@ -247,41 +250,69 @@ private async ValueTask<IVsWindowFrame> CreateToolWindowAsync(WorkspaceVisualNod
-            var caption = string.Format(CultureInfo.CurrentCulture, Resx.Label_NuGetWindowCaption, Path.GetFileNameWithoutExtension(workspaceVisualNodeBase.NodeMoniker));
+            IVsWindowFrame windowFrame = null;
 
             var uiShell = await _asyncServiceProvider.GetServiceAsync(typeof(SVsUIShell)) as IVsUIShell;
             Assumes.Present(uiShell);
 
-            int[] pfDefaultPosition = null;
-            IVsWindowFrame windowFrame;
-            var windowPane = new PackageManagerToolWindowPane(control);
-            ErrorHandler.ThrowOnFailure(
-                uiShell.CreateToolWindow(
-                    (uint)__VSCREATETOOLWIN.CTW_fInitNew,
-                    0,              // dwToolWindowId - singleInstance = 0
-                    windowPane,     // ToolWindowPane
-                    Guid.Empty,     // rclsidTool = GUID_NULL
-                    projectGuid,
-                    Guid.Empty,     // reserved - do not use - GUID_NULL
-                    null,           // IServiceProvider
-                    caption,
-                    pfDefaultPosition,
-                    out windowFrame));
-
-            if (windowFrame != null)
+            uint toolWindowId;
+            bool foundToolWindowId = _projectGuidToToolWindowId.TryGetValue(projectGuid.ToString(), out toolWindowId);
+            const uint FTW_none = 0;
+
+            if (foundToolWindowId)
             {
-                WindowFrameHelper.AddF1HelpKeyword(windowFrame, keywordValue: F1KeywordValuePmUI);
-                WindowFrameHelper.DisableWindowAutoReopen(windowFrame);
-                WindowFrameHelper.DockToolWindow(windowFrame);
+                uiShell.FindToolWindowEx(
+                    FTW_none, //grfFTW - badly-documented enum value.
+                    typeof(PackageManagerToolWindowPane).GUID,    // rguidPersistenceSlot
+                    toolWindowId,   // dwToolWindowId
+                    out windowFrame);
+                ((IVsWindowFrame2)windowFrame).ActivateOwnerDockedWindow();
+            }
+
+            if (windowFrame == null)
+            {
+                IProjectContextInfo projectContextInfo = await ProjectContextInfo.CreateAsync(projectGuid.ToString(), CancellationToken.None);
+                INuGetUI uiController = UIFactory.Value.Create(projectContextInfo);
+                var model = new PackageManagerModel(uiController, isSolution: false, editorFactoryGuid: GuidList.NuGetEditorType);
+                var control = await PackageManagerControl.CreateAsync(model, OutputConsoleLogger.Value);
+                var caption = string.Format(CultureInfo.CurrentCulture, Resx.Label_NuGetWindowCaption, Path.GetFileNameWithoutExtension(workspaceVisualNodeBase.NodeMoniker));
+
+                int[] pfDefaultPosition = null;
+
+                var windowPane = new PackageManagerToolWindowPane(control, projectGuid.ToString());
+                ErrorHandler.ThrowOnFailure(
+                    uiShell.CreateToolWindow(
```

Comment:
FindToolWindow can create the windowpane if not found, can/do we want to switch over to just the one call instead of CreateToolWindow?

Reply:
we still need to populate it, so most of the code in the if block would stay.
if we look at Document window code that we have, we have a hacky "find" implemented, and then have a CreateDocumentWindow call.
We can likely switch to that model, i don't feel the need strongly, but if you vote for that, i'd go do it.

Reply:
I don't feel strongly just wanted to know if we had considered it, if you think this is better thats fine.

Reply:
will play with it...

Reply:
played. didn't love the change, so staying here.